etcdserver/api/v3lock/v3lockpb/v3lock.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+package v3lockpb;
+
+import "gogoproto/gogo.proto";
+import "etcd/etcdserver/etcdserverpb/rpc.proto";
+
+// for grpc-gateway
+import "google/api/annotations.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+service Lock {
+  rpc Lock(LockRequest) returns (LockResponse) {
+      option (google.api.http) = {
+        post: "/v3alpha/lock/lock"
+        body: "*"
+    };
+  }
+  rpc Unlock(UnlockRequest) returns (UnlockResponse) {
```

Comment:
lets say we want to a lock with name `mylock`. the returned the key will be `mylock/001`. Then the client can check it `mylock/abc` still exists to see if the lock is still held by the owner? we have the guarantee that lock key is unique?